hardeneks/harden.py

```diff
@@ -1,17 +1,159 @@
 from importlib import import_module
 
+import sys
+import boto3
+from kubernetes import client
+from rich import print
 
-def harden(resources, config, _type):
+from rich.console import Console
+
+from .report import print_console_message
+
+from .report import colorMap
+
+def cluster_data(resources, config, _type):
+
+    current_eks_version = "1.24"
+    eks_cluster_data = []
+    
+    eksclient = boto3.client("eks", region_name=resources.region)
+    cluster_metadata = eksclient.describe_cluster(name=resources.cluster)
+    cluster_version = cluster_metadata["cluster"]["version"]
+    
+    if cluster_version == current_eks_version:
```

Comment:
should we make this more dynamic? As soon as 1.25 comes out this will be stale and need to be updated. The Best Practices say to use 1.24 or above. https://aws.github.io/aws-eks-best-practices/scalability/docs/control-plane/#use-eks-124-or-above so can we just check if current_version >= 1.24"